enhancements/microshift/advertise-version-details.md

```diff
@@ -0,0 +1,272 @@
+information.
+
+It is possible on first launch of a cluster for the API server to
+respond before the ConfigMap has been written, and therefore for the
+caller to fail to detect that the product is an OpenShift
+variant. This can be mitigated by populating the data as early as
+possible using an empty ConfigMap with only the `product` field filled
+in.
+
+There is a similar race condition when restarting MicroShift after an
+upgrade when the wrong version information may be presented. Clients
+are not as likely to need to query the MicroShift version at runtime
+in a production setting, because most deployments will either use
+manifests built into the image or use an agent to control application
+deployment from a central server. This race should therefore only
+affect developer scenarios.
+
+### Drawbacks
+
+None
+
+## Design Details
+
+### Open Questions
+
+1. Should we use a different namespace, like `kube-public`?
+2. Should we include any other information in the ConfigMap?
+3. Should `oc version` be extended to include this information?
+
+### Test Plan
+
+It should be possible to test this new behavior via functional tests
+without a full end-to-end test suite.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+See the workflow section above.
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+N/A
+
+## Implementation History
+
+https://github.com/openshift/microshift/pull/776 implemented an early
+version using a different ConfigMap in MicroShift.
+
+https://github.com/openshift/microshift/pull/814 updates the
+implementation in MicroShift to match this document.
+
+## Alternatives
+
+We could add the `ClusterVersion` API to MicroShift. This would add an
+API owned by a component not present in the cluster
+(cluster-version-operator), which would mean at least updating the
+MicroShift startup code to write to the API instead of a ConfigMap. It
+would also imply that a MicroShift cluster's version could be updated
+by using the API, which is not true.
+
+We could do nothing in OCP clusters. This would not solve the problem
+faced by the console of presenting version details to users who do not
+have admin privileges.
+
```

Comment:
does microshift include the `infrastructure` resource type?  I could see that being another place to put this information if microshift already has it.

Reply:
The list is extremely small and there are no APIs from the `config.openshift.io` space at all. The [list that is supported](https://github.com/openshift/microshift/blob/main/docs/enable_apis.md) includes only functional APIs, not configuration or informational APIs.